app/controllers/saml_idp_controller.rb

```diff
@@ -104,8 +104,15 @@ def saml_metadata
     SamlEndpoint.new(params[:path_year]).saml_metadata
   end
 
+  def ialmax_request_with_ial1_acr_and_pii_requested_and_locked?
+    requested_ial == 'ialmax' &&
```

Comment:
This method looks directly at the SAML request that is present here instead of at the AuthnContextResolver result so it is not affected by the 50/50 issue.